wallet.go

```diff
@@ -67,7 +67,7 @@ func Open(ctx context.Context,
 	wallet.timeout = parameters.timeout
 	wallet.endpoints = make([]*Endpoint, len(parameters.endpoints))
 	wallet.connectionProvider = &PuddleConnectionProvider{
-		name:            parameters.name,
+		name:            parameters.connectionName,
```

Comment:
`name` is mandatory but `connectionname` is not.  So I think that this should use `connectionName` if present, otherwise it should fall back to using `name`.